l5kit/l5kit/data/map_api.py

```diff
@@ -134,6 +142,53 @@ def get_lane_coords(self, element_id: str) -> dict:
 
         return {"xyz_left": xyz_left, "xyz_right": xyz_right}
 
+    @lru_cache(maxsize=CACHE_SIZE)
+    def get_lane_as_interpolation(self, element_id: str, step: float, method: InterpolationMethod) -> dict:
+        """
+        Perform an interpolation of the left and right lanes.
+        Two modes are available:
+        INTER_METER: interpolate using step as a meter value over cumulative distances (variable len result)
+        INTER_ENSURE_LEN: interpolate using a variable step such that we always get step values
+
+        Args:
+            element_id (str): lane id
+            step (float): step param for the method
+            method (InterpolationMethod): one of the accepted methods
+
+        Returns:
+            dict: same as `get_lane_coords` but overwrite xyz values for the lanes
+        """
+        lane_dict = self.get_lane_coords(element_id)
+        for lane_key in ["xyz_left", "xyz_right"]:
+            xyz_raw = lane_dict[lane_key]
+
+            # cumulative distance from the first point, including the first itself as 0
+            distances = np.cumsum(np.linalg.norm(np.diff(xyz_raw, axis=0), axis=-1))
+            distances = np.insert(distances, 0, 0)
```

Comment:
Wondering if "insert" in idx=0 is costly here, since it may imply moving all elements... tried to look up but couldn't get a good answer. Do you think they maybe a concatenation, or initializing distances with all zeros could be better? I may be way off though.

Reply:
I think (and I would like to stress the "think") that numpy stores doubled linked list here. Anyway, this function is LRU cached, so it should be fine :)

Reply:
That's great then.